community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/plans/LogicalPlan.scala

```diff
@@ -68,6 +68,16 @@ abstract class LogicalPlan
     }
   }
 
+  def copyPlan(): LogicalPlan = {
+    try {
+      val arguments = this.children.toList :+ solved
+      copyConstructor.invoke(this, arguments: _*).asInstanceOf[this.type]
```

Comment:
Could we reuse the `dup` method in `DuplicatableAny` to at least avoid referencing the copy constructor in multiple places?